### PR TITLE
issue #89, container_service: implement ecs cluster

### PR DESCRIPTION
**Pull request by icahoon**

### CODE REVIEW

**Review by gstragand — APPROVED** [review]
lgtm

**Review by danielcahoon — APPROVED** [review]
lgtm

File: pkg/mock/container_service.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+func newContainerService(cfg *config.ContainerService, p *containerServiceProvider) (resource.ProviderContainerService, error) {
+	log.Info("Initializing Mock Container Service")
+	cs := &containerService{
+		ContainerService: cfg,
+		opt:              options{p.Provider.Data},
+	}
+	if cs.opt.err("cs.New") {
+		return nil, dberr{"cs.New"}
```
Comment (danielcahoon) [inline pkg/mock/container_service.go]: reuse of dberr instead of an error specifically for cs

File: pkg/mock/container_service.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	}
+	return cs, nil
+}
+
+func (cs *containerService) Load() error {
+	log.Info("Loading Mock Container Service")
+	if cs.opt.err("cs.Load") {
+		return dberr{"cs.Load"}
```
Comment (danielcahoon) [inline pkg/mock/container_service.go]: reuse of dberr instead of an error specifically for cs

File: pkg/mock/container_service.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	}
+	return nil
+}
+
+func (cs *containerService) Create(flags ...string) error {
+	msg.Info("Creating Mock ContainerService")
+	if cs.opt.err("cs.Create") {
+		return dberr{"cs.Create"}
```
Comment (danielcahoon) [inline pkg/mock/container_service.go]: reuse of dberr instead of an error specifically for cs

File: pkg/mock/container_service.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	if cs.opt.err("cs.Created") {
+		return false
+	}
+	return true
+}
+
+func (cs *containerService) Destroy(flags ...string) error {
+	msg.Info("Destroying Mock ContainerService")
+	if cs.opt.err("cs.Destroy") {
+		return dberr{"cs.Destroy"}
```
Comment (danielcahoon) [inline pkg/mock/container_service.go]: reuse of dberr instead of an error specifically for cs

File: pkg/mock/container_service.go [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+	if cs.opt.err("cs.Destroyed") {
+		return false
+	}
+	return true
+}
+
+func (cs *containerService) Provision(flags ...string) error {
+	msg.Info("Provisioning Mock ContainerService")
+	if cs.opt.err("cs.Provision") {
+		return dberr{"cs.Provision"}
```
Comment (danielcahoon) [inline pkg/mock/container_service.go]: reuse of dberr instead of an error specifically for cs

**Review by danielcahoon — APPROVED** [review]

**Comment by icahoon** [comment]
closes issue #89

**Pull request merged**